src/BizHawk.Emulation.Cores/Consoles/Nintendo/Gameboy/Gambatte.IEmulator.cs

```diff
@@ -9,7 +9,7 @@ public partial class Gameboy : IEmulator, IBoardInfo
 	{
 		public IEmulatorServiceProvider ServiceProvider { get; }
 
-		public ControllerDefinition ControllerDefinition => (_syncSettings.FrameLength == GambatteSyncSettings.FrameLengthType.UserDefinedFrames) ? SubGbController : GbController;
+		public ControllerDefinition ControllerDefinition => CreateControllerDefinition(IsSgb, _syncSettings.FrameLength is GambatteSyncSettings.FrameLengthType.UserDefinedFrames);

```

Comment:
Is `FrameLength` allowed to change while the core is running? I assume not, so this only needs to be called once during initialisation.
```suggestion
		public ControllerDefinition ControllerDefinition { get; } = CreateControllerDefinition(IsSgb, _syncSettings.FrameLength is GambatteSyncSettings.FrameLengthType.UserDefinedFrames);
```

Reply:
It technically can change, although it would probably be better anyways if it doesn't take affect until core reboot. Although that change doesn't actually build. ("A field initializer cannot reference the non-static field, method, or property 'Gameboy.IsSgb'")